### PR TITLE
common - Add pldm cpld update and improve pldm update common code

### PR DESCRIPTION
**Pull request by MouchenHung-QUANTA**
Summary:
- Support cpld pldm fw update
- Support vr update for Grand Teton

Test Plan:
- Build Code: PASS

### CODE REVIEW

**Comment by MouchenHung-QUANTA** [comment]
Hi @garnermic , This pull request is based on #780 , thanks!

File: common/service/pldm/pldm_firmware_update.c [hunk 1]
```diff
@@ -37,147 +41,144 @@ k_tid_t fw_update_tid;
...
+		}
+		hex_buff = malloc(fw_update_cfg.image_size);
+		if (!hex_buff) {
+			LOG_ERR("Failed to malloc hex_buff");
 			return 1;
 		}
+	}
 
-		memcpy(cur_p, &resp_buf[1], req.length);
+	memcpy(hex_buff + p->data_ofs, p->data, p->data_len);
```
Comment (garnermic) [inline common/service/pldm/pldm_firmware_update.c:118]: if p->data_ofs is not zero, then hex_buff is NULL here.

**Review by garnermic — CHANGES_REQUESTED** [review]

**Comment by MouchenHung-QUANTA** [comment]
Hi @garnermic , New commit has fix the problem, thanks for remind.

**Review by garnermic — CHANGES_REQUESTED** [review]
Request was made to have the Github actions addressed before Meta approval.

**Comment by garnermic** [comment]
The latest update still shows it is failing Github checks as shown below.

Build and Analyze / run-cppcheck
Build and Analyze / run-clang-format
Build and Analyze / Aggregate-Lint-Output

**Comment by MouchenHung-QUANTA** [comment]
Hi @garnermic , merge conflict problem has been solved, please have a look, thanks!

**Comment by MouchenHung-QUANTA** [comment]
Since #800 has been merged, so closed this pull request.

**Pull request closed without merging**